packages/common/src/groups/_internal/GroupBusinessRules.ts

```diff
@@ -80,6 +81,23 @@ export const GroupPermissionPolicies: IPermissionPolicy[] = [
     authenticated: true,
     entityOwner: true,
   },
+  {
+    permission: "hub:group:canEditAccess",
+    dependencies: ["hub:group"],
+    authenticated: true,
+    assertions: [
+      {
+        property: "context:currentUser.privileges",
+        type: "contains-some",
+        value: ["portal:admin:updateGroups"],
+      },
+      {
+        property: "context:currentUser",
+        type: "is-group-admin",
+        value: "entity:id",
+      },
```

Comment:
I may be reading the issue wrong, so totally correct me if I'm off here -- but Jay said this towards the bottom: 

> Hence, I think group sharing should be limited to (userMembership.membershipType: "admin" OR userMembership.membershipType: "owner" OR portal:admin:updateGroups)

I think, with how we handle assertions, your addition here currently reads: 
`(userMembership.membershipType: "admin" OR userMembership.membershipType: "owner") AND portal:admin:updateGroups`

Again, might be totally off here, but I think we'd need to combine these assertions in a way so that we have them joined by or's. Which all gets really complex 😅 I'm not sure that would be possible right now without adding something new, like an operator for all of the assertions....or creating separate permissions. 

Can a user even have the `updateGroups` permission without being an admin or an owner?

Reply:
Oh man, nice catch - I definitely missed that last `OR`. And that does significantly complicate this 😅 
I'm trying to think if I could use conditions in any way on these assertions, but I don't _think_ that gets us all the way. Hmmm, being able to specify an operator is a good thought, but that does seem like a heavier lift here. 

> Can a user even have the updateGroups permission without being an admin or an owner?
I think the answer is technically yes, even though that doesn't make too much sense, but I think we have to account for it. 

creating separate permissions is probably the easiest solution for now, but I don't love that - it would be nice if all entities had this same permission to key off of. I'll keep pondering

Reply:
just spitballing, so I could be way off here, but with conditions, wondering if we'd have to do something like this: 

```ts
{
  permission: "hub:group:canEditAccess",
  dependencies: ["hub:group"],
  authenticated: true,
  assertions: [
    {
      conditions: [
        {
          property: "context:currentUser",
          type: "is-not-group-admin", // this type doesn't currently exist but would be easy to add and would basically just check if you're a "member"
          value: "entity:id",
        },
      ],
      property: "context:currentUser.privileges",
      type: "contains",
      value: ["portal:admin:updateGroups"],
    },
    {
      conditions: [
        {
          property: "context:currentUser.privileges",
          type: "without",
          value: ["portal:admin:updateGroups"],
        }
      ],
      property: "context:currentUser",
      type: "is-group-admin",
      value: "entity:id",
    },
  ],
},
```

Still parsing this myself, and it does seem kind of redundant, but does this give us what we need? 😅

Reply:
I think so -- that was along the idea I was playing around with yesterday as well! 

The first assertion is: if someone isn't an admin, do they have the updateGroups priv?
The second assertion is: if someone doesn't have the updateGroups priv, are they an admin?

If someone is an admin && has the updateGroups priv, neither assertion would be "checked"...but I think that's okay, because in that case, the permission would come back with access: true, since all assertions (none) pass...right?

Reply:
yes, that's how I read it as well and think that all sounds correct!